pkgs/applications/editors/emacs/generic.nix

```diff
@@ -391,44 +392,7 @@ mkDerivation (finalAttrs: (lib.optionalAttrs withNativeCompilation {
     treeSitter = builtins.trace "emacs.passthru: treeSitter was renamed to withTreeSitter and will be removed in 23.11" withTreeSitter;
   };
 
-  meta = {
-    homepage = if variant == "macport"
-               then "https://bitbucket.org/mituharu/emacs-mac/"
-               else "https://www.gnu.org/software/emacs/";
-    description = "The extensible, customizable GNU text editor"
-                  + lib.optionalString (variant == "macport") " - with macport patches";
-    longDescription = ''
-      GNU Emacs is an extensible, customizable text editor—and more. At its
-      core is an interpreter for Emacs Lisp, a dialect of the Lisp programming
-      language with extensions to support text editing.
-
-      The features of GNU Emacs include: content-sensitive editing modes,
-      including syntax coloring, for a wide variety of file types including
-      plain text, source code, and HTML; complete built-in documentation,
-      including a tutorial for new users; full Unicode support for nearly all
-      human languages and their scripts; highly customizable, using Emacs Lisp
-      code or a graphical interface; a large number of extensions that add other
-      functionality, including a project planner, mail and news reader, debugger
-      interface, calendar, and more. Many of these extensions are distributed
-      with GNU Emacs; others are available separately.
-    ''
-    + lib.optionalString (variant == "macport") ''
-
-      This release is built from Mitsuharu Yamamoto's patched, MacOS X-specific
-      source code.
-    '';
-    license = lib.licenses.gpl3Plus;
-    maintainers = with lib.maintainers; [
-      AndersonTorres
-      adisbladis
-      atemu
-      jwiegley
-      lovek323
-      matthewbauer
-    ];
-    platforms = if variant == "macport"
-                then lib.platforms.darwin
-                else lib.platforms.all;
+  meta = meta // {
     broken = !(stdenv.buildPlatform.canExecute stdenv.hostPlatform);
```

Comment:
Should I put this on the meta too? It would force the inclusion of stdenv onto sources logic, tho.

Reply:
It's probably better where it is as it's completely generic.
Let's keep the sources logic as tiny as possible.